hack/update-proto.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+SOURCE="${BASH_SOURCE[0]}"
+while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
+  DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+  SOURCE="$(readlink "$SOURCE")"
+  [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
+done
+ROOT_DIR="$( cd -P "$( dirname "$SOURCE" )/.." && pwd )"
+
+mkdir -p "${ROOT_DIR}/bin"
+export GOBIN=$ROOT_DIR/bin
+
+if [ ! -f "${GOBIN}/buf" ]; then 
+  go install github.com/bufbuild/buf/cmd/buf@v1.32.2
```

Comment:
Do we want to use go install to install buf or we can use `brew` as explained here: https://github.com/bufbuild/buf?tab=readme-ov-file#homebrew ?

Reply:
I would like to keep using this approach as opposed to `brew` since I don't want to add new tools for the developers.

https://github.com/kubeflow/katib/blob/master/docs/developer-guide.md#requirements

Reply:
Sounds good.